### PR TITLE
fix(tokenizer): Parse entities in <title> tags

### LINKED ISSUE #592: <title> tag content will never decode html entities
0189e56a876e17c60be5ff7433bcfea3756f9ab9 introduced a bug where html entities inside <title> tags are not decoded, even when the parser is set to decode entities.

Example test code:
```js
// test.js
const htmlparser2 = require("./lib/index");
const parser = new htmlparser2.Parser({
    ontext(text) {
        console.log("-->", text);
    },
}, { decodeEntities: true });
parser.write("<title>my&quot;title&quot;");
parser.end();
```

Before 0189e56a876e17c60be5ff7433bcfea3756f9ab9:
```sh
$ node test.js
--> my
--> "
--> title
--> "
```

After 0189e56a876e17c60be5ff7433bcfea3756f9ab9:
```sh
$ node test.js
--> my&quot;title&quot;
```

This appears to be the result of trying to fix #482, however html entity decoding should still occur inside of title tags (even if additional html tags before the closing </title> should not)

### PR DESCRIPTION
**Pull request by fb55**
Fixes #592.

### CODE REVIEW

**Comment by fb55** [comment]
Beat you to it, hope that's alright!

@billneff79 @cheeseandcereal Would love a review if you guys have a moment.

File: src/__fixtures__/Events/37-entity-in-title.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        {
+            "event": "opentag",
+            "data": ["title", {}]
+        },
+        {
+            "event": "text",
+            "data": ["the \"title\""]
+        },
+        {
+            "event": "closetag",
```
Comment (cheeseandcereal) [inline src/__fixtures__/Events/37-entity-in-title.json:18]: is this event actually fired? I don't see a `</title>` in the html string for this fixture

Comment (fb55) [reply src/__fixtures__/Events/37-entity-in-title.json:18]: The parser will close all currently open tags at the end of the input.

**Review by cheeseandcereal — APPROVED** [review]
pulled and tested locally, confirming fix. lgtm

**Comment by fb55** [comment]
Thanks a lot @cheeseandcereal!

**Comment by billneff79** [comment]
> Beat you to it, hope that's alright!
> 
> @billneff79 @cheeseandcereal Would love a review if you guys have a moment.

Awesome!  Thanks @fb55 .  My only thought is that it would be tidy and safe to have one more unit test to verify that title does _not_ decode entities when `decodeEntities=false`.  As my humble contribution to this fix I've added that test fixture in #615

**Pull request merged**